inc/Engine/Media/ImageDimensions/Subscriber.php

```diff
@@ -64,6 +64,23 @@ public function specify_image_dimensions( $buffer ) {
 		return $this->dimensions->specify_image_dimensions( $buffer );
 	}
 
+	/**
+	 * Update images that have no width/height with real dimentions for the SaaS
+	 *
+	 * @param string $buffer Page HTML content.
+	 *
+	 * @return string Page HTML content after update.
+	 */
+	public function prepare_critical_image_saas_visit( $buffer ) {
+		if ( ! isset( $_GET['wpr_imagedimensions'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return $buffer;
+		}
+
+		do_action( 'rocket_critical_image_saas_visit_buffer', $buffer );
+
+		return $this->dimensions->specify_image_dimensions( $buffer );
+	}
```

Comment:
```suggestion
		return apply_filters( 'rocket_critical_image_saas_visit_buffer', $buffer );
	}
```

and in the `get_subscribed_events()`, add:
 `'rocket_critical_image_saas_visit_buffer' => 'specify_image_dimensions'`